build-system/tasks/release/index.js

```diff
@@ -0,0 +1,484 @@
+  '11': {type: 'experimentB', configBase: 'prod'},
+  '12': {type: 'experimentC', configBase: 'prod'},
+  '20': {type: 'experimentA-control', configBase: 'prod'}, // Spec name: 'inabox-experimentA-control'
+  '21': {type: 'experimentA', configBase: 'prod'}, // Spec name: 'inabox-experimentA'
+  '22': {type: 'experimentB-control', configBase: 'prod'}, // Spec name: 'inabox-experimentB-control'
+  '23': {type: 'experimentB', configBase: 'prod'}, // Spec name: 'inabox-experimentB'
+  '24': {type: 'experimentC-control', configBase: 'prod'}, // Spec name: 'inabox-experimentC-control'
+  '25': {type: 'experimentC', configBase: 'prod'}, // Spec name: 'inabox-experimentC'
+};
+
+// Mapping of entry file names to a dictionary of AMP_CONFIG additions.
+const TARGETS_TO_CONFIG = MINIFIED_TARGETS.flatMap((minifiedTarget) => [
+  {file: `${minifiedTarget}.js`, config: {}},
+  {file: `${minifiedTarget}.mjs`, config: {esm: 1}},
+]);
+
+function logSeparator_() {
+  log('---\n\n');
+}
+
+/**
+ * Prepares output and temp directories, and ensures dependencies are installed.
+ *
+ * @param {string} outputDir full directory path to emplace artifacts in.
+ * @param {string} tempDir full directory path to temporary working directory.
+ */
+async function prepareEnvironment_(outputDir, tempDir) {
+  await fs.emptyDir(outputDir);
+  await fs.emptyDir(tempDir);
+  logSeparator_();
+
+  execOrDie('gulp update-packages');
+  logSeparator_();
+}
+
+/**
+ * Discovers which AMP flavors are defined in the current working directory.
+ *
+ * The returned list of flavors will always contain the base flavor, and any
+ * defined experiments in ../../global-configs/experiments-config.json.
+ *
+ * @return {!Array<!Object>} list of AMP flavors to build.
+ */
+function discoverDistFlavors_() {
+  const distFlavors = [
+    BASE_FLAVOR_CONFIG,
+    ...Object.entries(experimentsConfig)
+      .filter(([, experimentConfig]) => experimentConfig.command)
+      .map(([flavorType, experimentConfig]) => ({
+        flavorType,
+        rtvPrefixes: [
+          EXPERIMENTAL_RTV_PREFIXES[experimentConfig.environment][flavorType],
+        ],
+        ...experimentConfig,
+      })),
+  ];
+  // TODO(#28168, danielrozenberg): if --flavor is defined, filter out the other
+  // flavors.
+
+  log(
+    'The following',
+    cyan('gulp dist'),
+    'commands will be executed to compile each',
+    `${green('flavor')}:`
+  );
+  distFlavors.forEach(({flavorType, name, command}) => {
+    log('-', `(${green(flavorType)}, ${green(name)})`, cyan(command));
+  });
+
+  logSeparator_();
+
+  return distFlavors;
+}
+
+/**
+ * Compiles all AMP flavors sequentially.
+ *
+ * @param {!Array<!Object>} distFlavors list of AMP flavors to build.
+ * @param {string} tempDir full directory path to temporary working directory.
+ */
+async function compileDistFlavors_(distFlavors, tempDir) {
+  for (const {flavorType, command} of distFlavors) {
+    log('Compiling flavor', `${green(flavorType)}:`);
```

Comment:
For debugging, can you also log `command` here?

Reply:
Done